pkg/network/node/metrics.go

```diff
@@ -0,0 +1,114 @@
+package node
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	SDNNamespace          = "openshift"
+	SDNSubsystem          = "sdn"
+	OVSFlowsKey           = "ovs_flows"
+	ARPCacheEntriesKey    = "arp_cache_entries"
+	PodIPsKey             = "pod_ips"
+	PodSetupErrorsKey     = "pod_setup_errors"
+	PodSetupLatencyKey    = "pod_setup_latency"
+	PodTeardownErrorsKey  = "pod_teardown_errors"
+	PodTeardownLatencyKey = "pod_teardown_latency"
+)
+
+var (
+	OVSFlows = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: SDNNamespace,
+			Subsystem: SDNSubsystem,
+			Name:      OVSFlowsKey,
+			Help:      "Number of Open vSwitch flows",
+		},
+	)
+
+	ARPCacheEntries = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: SDNNamespace,
+			Subsystem: SDNSubsystem,
+			Name:      ARPCacheEntriesKey,
+			Help:      "Number of entries in the ARP cache",
+		},
+	)
+
+	PodIPs = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: SDNNamespace,
+			Subsystem: SDNSubsystem,
+			Name:      PodIPsKey,
+			Help:      "Number of allocated pod IPs",
+		},
+	)
+
+	PodSetupErrors = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: SDNNamespace,
+			Subsystem: SDNSubsystem,
+			Name:      PodSetupErrorsKey,
+			Help:      "Number pod setup errors",
+		},
+	)
+
+	PodSetupLatency = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Namespace: SDNNamespace,
+			Subsystem: SDNSubsystem,
+			Name:      PodSetupLatencyKey,
+			Help:      "Latency of pod network setup in microseconds",
+		},
+		[]string{"sandbox_id"},
+	)
+
+	PodTeardownErrors = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: SDNNamespace,
+			Subsystem: SDNSubsystem,
+			Name:      PodTeardownErrorsKey,
+			Help:      "Number pod teardown errors",
+		},
+	)
+
+	PodTeardownLatency = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Namespace: SDNNamespace,
+			Subsystem: SDNSubsystem,
+			Name:      PodTeardownLatencyKey,
+			Help:      "Latency of pod network teardown in microseconds",
+		},
+		[]string{"sandbox_id"},
```

Comment:
Do we need the sandbox id to debug?  If not pod_name and namespace would be better because then we can correlate back to the affected pod.  Cardinality would be about the same.

Reply:
@smarterclayton kube sandbox operations are per-sandbox, not per-pod.  A pod could have multiple sandboxes (though only one is actually the 'running' sandbox that the pod actually uses).  The others are waiting to be garbage collected/torn down, which could happen later depending on kubelet's GC goroutine.  So these operations are inherently per-sandbox, not per-pod.